docs/preview/yamlgettingstarted-directory.md

```diff
@@ -0,0 +1,67 @@
+# YAML getting started - YAML directory (not available, for discussion only)
+
+## Summary
+
+We plan to support a convention-based directory structure, with a couple goals in mind:
+
+1. Today we only create a build definition on-push for the root file `.vsts-ci.yml`. And resource authorization on-push is only handled for that one file. With a convention based
+directory structure, we can watch for files under a specific directory.
+
+2. For template scenarios, a convention-based directory structure would enable referencing templates by name, instead of by relative path.
+
+## Directory structure
+
+```
+.vsts-pipelines/
+  builds/
+    build.yml
+  releases/
+    release.yml
+  steps/
+    msbuild.yml
+    node.yml
+  phases/
+  stages/
+```
+
+## Definition creation on-push
+
+On push, builds or releases would be created for yml files under the `builds` and `releases` directories.
```

Comment:
Is there a way to configure the directory to look under?

Reply:
@weshaggard - no.  And that's very intentional.  We need to scope our search for large pushes.

Reply:
I understand the desire to limit the scope but it still feels like this could be configurable at some level (repo/definition/vsts instance). There are reasons for different projects to need a different set of directory structure.

Reply:
Interested in those scenarios ... feel free to email/discuss with @chrisrpatterson and I as well if that's easier.

There's also a desire for approachability of a repo where you can approach the repro and instantly what it's process is (without access to the account's config).  We're also considering marketing and names in this.  (vsts-yaml is instantly recognizable, considering for folder as well)

Reply:
Is it possible to maintain the above described folder structure, but move the entire structure to some other directory and search for the existence of the `.vsts-pipelines/` folder?  So, your search is to find `.vsts-pipelines`, and then do all the yml search logic.  That doesn't address the approachability aspect though.

Another option would be to have a root `.vsts-config.yml` file